src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/ForeignIslandTableManager.java

```diff
@@ -26,18 +26,19 @@ public ForeignIslandTableManager(ConnectionSource connectionSource, Class<T> cla
 
     @Override
     public void addEntry(T t) {
-        getEntries().add(t);
+        int index = Collections.binarySearch(getEntries(), t, Comparator.comparing(t1 -> t1.getIsland().map(Island::getId).orElse(0)));
```

Comment:
This shouldnt be needed, it should add it in its correctly sorted place if you look in TableManager and SortedList class